packages/nc-gui/composables/useProject.ts

```diff
@@ -104,8 +106,15 @@ const [setup, use] = useInjectionState((_projectId?: MaybeRef<string>) => {
     if (id) {
       project.value = await api.project.read(projectId.value)
     } else if (projectType === 'base') {
-      const baseData = await api.public.sharedBaseGet(route.params.projectId as string)
-      project.value = await api.project.read(baseData.project_id!)
+      try {
+        const baseData = await api.public.sharedBaseGet(route.params.projectId as string)
+        project.value = await api.project.read(baseData.project_id!)
+      } catch (e: any) {
+        if (e?.response?.status === 404) {
+          return router.push('/error/404')
```

Comment:
Instead of using router.push there is a dedicated function by nuxt for this:
https://v3.nuxtjs.org/getting-started/error-handling#createerror

Would that not be better? Includes status code etc. This would push the 404 as a Status 200 page, wouldn't it?

Reply:
Not available in the current version we are using, we can refactor once we migrate to the latest nuxt